Automattic/wp-calypso
# Site Switcher: Apply Sites page sorting to items

#### Proposed Changes

Closes https://github.com/Automattic/wp-calypso/issues/67043.

This PR applies the same sorting mechanisms used in `/sites` on all instances of the site selector. This is so we can have consistency between lists and pages.

This also removes the `recentSites` preference dependency as we now want to have the same order across all site lists. The preference is still updated whenever a site is visited as there are a lot of use cases where this preference is read.

#### Testing Instructions

1. Open the Sites page and change the sorting order;
2. Take note of the list ordering;
3. Open the Site Switcher and check that the items are ordered equally.
